### ISSUE TITLE
Whole traceback thrown when subproc'ing to a non-existing command

### ISSUE THREAD

**Issue by hfaran** [scopatz/xonsh#54]
This is how bash handles it:
```bash
^_^hfaran@_ ~
$ foobar
-bash: foobar: command not found
```

And here is xonsh:
```python
hfaran@_ ~ $ foobar
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/xonsh/shell.py", line 109, in default
    self.execer.exec(code, mode='single', glbs=None, locs=self.ctx)
  File "/usr/local/lib/python3.4/site-packages/xonsh/execer.py", line 110, in exec
    return exec(code, glbs, locs)
  File "<xonsh-code>", line 1, in <module>
  File "/usr/local/lib/python3.4/site-packages/xonsh/built_ins.py", line 383, in subproc_uncaptured
    return run_subproc(cmds, captured=False)
  File "/usr/local/lib/python3.4/site-packages/xonsh/built_ins.py", line 355, in run_subproc
    stdin=stdin, stdout=stdout)
  File "/usr/local/Cellar/python3/3.4.3/Frameworks/Python.framework/Versions/3.4/lib/python3.4/subprocess.py", line 859, in __init__
    restore_signals, start_new_session)
  File "/usr/local/Cellar/python3/3.4.3/Frameworks/Python.framework/Versions/3.4/lib/python3.4/subprocess.py", line 1457, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'foobar'
```

Ideally, I think it would be preferable not to see the raw exception.

**Comment by scopatz** [comment]
It probably would be, I agree. Thanks for pointing this out @hfaran.  Pull requests welcome, of course.

**Comment by hfaran** [comment]
I would be more than happy to submit a pull request for this (whenever I find the time, which may be later rather than sooner). That being said, in your opinion, what would be good behaviour here (or good output)?

On the one hand, there is the argument, that since is xonsh is actually a Python superset, the whole stacktrace should be shown. On the other hand, xonsh is also a shell, and seeing that stacktrace every time you enter a command that cannot be found presents a usability issue.

I am of the opinion that the internals of a shell should never be exposed, i.e., if a stacktrace is thrown, that means something unexpected has happened and is a bug in the shell.

So for a shell like xonsh, while I consider a `FileNotFoundError` propogating to the top in this context unacceptable, I do consider two things acceptable:
* An exception being raised, still, however, it should abstract away the internals of xonsh and really should be an abstraction layer, i.e., part of the xonsh API, i.e., a custom xonsh exception.
* More bash-like behaviour where a simple one-line error is printed to STDERR.

I look forward to hearing your opinions on this.

**Comment by mbollmann** [comment]
My 5 cents: I basically agree, though note that you might only want to hide stacktraces that relate to xonsh internals, i.e., if you define and execute a Python function in the shell and it throws, I'd not find it unusual to see the stacktrace. Maybe that was what you intended anyway, I just wanted to make it explicit.

I think this should ideally be made into its own "error handling" class, so you don't have raw error strings floating around everywhere, and you could optionally turn the stacktraces back on (for debugging).

**Comment by novist** [comment]
This includes ``KeyboardInterrupt`` error. Ctrl-c atm produces lots of noise.

**Comment by hfaran** [comment]
Yes, @novist, I also noticed that but I believe it is different enough to deserve its own Issue.

**Comment by novist** [comment]
Well it still comes from ``execer.py``

**Comment by hfaran** [comment]
Well, for this issue, you would actually need to look `xonsh.built_ins.run_subproc` first and then look at how it interfaces with `xonsh.execer`, but my point was is that handling Ctrl-C in the form of a `KeyboardInterrupt` needs to be done elsewhere.

**Comment by scopatz** [comment]
fixed by #58
